inc/zoo/swar/SWAR.h

```diff
@@ -62,18 +73,82 @@ struct SWAR {
         AllOnes = ~std::make_unsigned_t<T>{0} >> PaddingBitsCount, // Also constructed in RobinHood utils: possible bug?
         LeastSignificantBit = meta::BitmaskMaker<T, std::make_unsigned_t<T>{1}, NBits>::value,
         MostSignificantBit = LeastSignificantBit << (NBits - 1),
-        LeastSignificantLaneMask =
-            sizeof(T) * 8 == NBits ? // needed to avoid shifting all bits
-                ~T(0) :
-                ~(~T(0) << NBits),
+        LeastSignificantLaneMask = []() {
+            if constexpr (NBits < sizeof(T) * 8) {
+                return (T(1) << NBits) - 1;
+            } else {
+                return ~T(0);
+            }
+        }(),
         // Use LowerBits in favor of ~MostSignificantBit to not pollute
         // "don't care" bits when non-power-of-two bit lane sizes are supported
-        LowerBits = MostSignificantBit - LeastSignificantBit;
+        LowerBits = MostSignificantBit - LeastSignificantBit,
+        MaxUnsignedLaneValue = LeastSignificantLaneMask;
+
+    template <typename InputIt>
+    constexpr static auto from_range(InputIt first, InputIt last) noexcept {
+        auto result = T{0};
+        for (; first != last; ++first) {
+            result = (result << NBits) | *first;
+        }
+        return result;
+    }
+
+    template <typename U>
+    constexpr static auto from_array(const U (&values)[Lanes]) noexcept {
+        using std::begin; using std::end;
+        return SWAR{from_range(begin(values), end(values))};
+    }
+
+    template <typename U>
+    constexpr static auto from_array(const std::array<T, Lanes> &values) noexcept {
+        using std::begin; using std::end;
+        return SWAR{from_range(begin(values), end(values))};
+    }
+
+    using ArrayType = std::array<T, Lanes>;
+
+    constexpr SWAR(const ArrayType &array) : m_v{from_range(array.begin(), array.end())} {}
+
+    template <typename Arg, std::size_t N, typename = std::enable_if_t<N == Lanes, int>>
+    constexpr
+    SWAR(Literals_t<NBits, T>, const Arg (&values)[N]) : m_v{from_array(values)} {}
+
+    constexpr std::array<T, Lanes> to_array() const noexcept {
+        std::array<T, Lanes> result = {};
+        for (int i = 0; i < Lanes; ++i) {
+            auto otherEnd = Lanes - i - 1;
+            result[otherEnd] = at(i);
+        }
+        return result;
```

Comment:
I think this can be encapsulated into an expression

Reply:
you think this should be in some named expression like `convertSwarToArray` ?

Reply:
I did not understadn your question, this is not about a name or identifier, but something like
```
template<yada yada>
some_converter_name_impl(std::index_sequence<Indices...>, arguments) {
    return { at(Indices)...};
}
```
where `Indices` is an argument pack of numbers 0 to the lane count

Reply:
Not really following what you're getting at here? Would this be for going to or from an array?  
We could substitude doing the weird array packs we're been doing and use `integer_sequence` instead, possibly?

Reply:
Leave it as it is, if you want.
I will code from memory, something like this:
namespace impl {
template<typename T, std::size_t... Indices>
constexpr
std::array<T, sizeof...(Indices)> to_array(SWAR<?????> s, std::index_sequence<Indices...>) noexcept {
    return { s.at(Indices)... };
}
}
// ...
constexpr std::array<T, Lanes> to_array() const noexcept {
    return impl::to_array(*this, std::make_index_sequence<Lanes>{});
}
```
I've always thought of this idiom, that to give it a name I called it "expression style" as more elegant; I love translating the minutiae of iteration to compile-time-only code.
I also dislike the imperative style.

Reply:
Let's do this in another PR.
I think Scott would like to start using the ergonomics in writing some tests. 
It will be valueable to let Scott's magnetic field to push back on this code's electrical field.